contrib/vcl/source/vcl_io_handle.h

```diff
@@ -0,0 +1,103 @@
+                                  const Envoy::Network::Address::Instance& peer_address) override;
+  Api::IoCallUint64Result recvmsg(Buffer::RawSlice* slices, const uint64_t num_slice,
+                                  uint32_t self_port, RecvMsgOutput& output) override;
+  Api::IoCallUint64Result recvmmsg(RawSliceArrays& slices, uint32_t self_port,
+                                   RecvMsgOutput& output) override;
+  absl::optional<std::chrono::milliseconds> lastRoundTripTime() override;
+
+  bool supportsMmsg() const override;
+  bool supportsUdpGro() const override { return false; }
+
+  Api::SysCallIntResult bind(Envoy::Network::Address::InstanceConstSharedPtr address) override;
+  Api::SysCallIntResult listen(int backlog) override;
+  Envoy::Network::IoHandlePtr accept(struct sockaddr* addr, socklen_t* addrlen) override;
+  Api::SysCallIntResult connect(Envoy::Network::Address::InstanceConstSharedPtr address) override;
+  Api::SysCallIntResult setOption(int level, int optname, const void* optval,
+                                  socklen_t optlen) override;
+  Api::SysCallIntResult getOption(int level, int optname, void* optval, socklen_t* optlen) override;
+  Api::SysCallIntResult ioctl(unsigned long control_code, void* in_buffer,
+                              unsigned long in_buffer_len, void* out_buffer,
+                              unsigned long out_buffer_len, unsigned long* bytes_returned) override;
+  Api::SysCallIntResult setBlocking(bool blocking) override;
+  absl::optional<int> domain() override;
+  Envoy::Network::Address::InstanceConstSharedPtr localAddress() override;
+  Envoy::Network::Address::InstanceConstSharedPtr peerAddress() override;
+  Api::SysCallIntResult shutdown(int) override { return {0, 0}; }
+
+  void initializeFileEvent(Event::Dispatcher& dispatcher, Event::FileReadyCb cb,
+                           Event::FileTriggerType trigger, uint32_t events) override;
+  void activateFileEvents(uint32_t events) override { file_event_->activate(events); }
+  void enableFileEvents(uint32_t events) override { file_event_->setEnabled(events); }
+  void resetFileEvents() override { file_event_.reset(); }
+  IoHandlePtr duplicate() override;
+
+  void cb(uint32_t events) { cb_(events); }
+  void setCb(Event::FileReadyCb cb) { cb_ = cb; }
+  void updateEvents(uint32_t events);
+  uint32_t sh() const { return sh_; }
+  void clearChildWrkListener() {
+    if (wrk_listener_) {
+      wrk_listener_ = nullptr;
+    }
+  }
+  VclIoHandle* getParentListener() { return parent_listener_; }
+  bool isWrkListener() { return parent_listener_ != nullptr; }
+
+private:
+  void setChildWrkListener(VclIoHandle* parent_listener) { parent_listener_ = parent_listener; }
+
+  uint32_t sh_{VCL_INVALID_SH};
+  os_fd_t fd_;
+  Event::FileEventPtr file_event_{nullptr};
+  bool is_listener_ = false;
+  bool not_listened_ = false;
+  VclIoHandle* parent_listener_{nullptr};
+  std::unique_ptr<VclIoHandle> wrk_listener_{nullptr};
```

Comment:
Add a comment on what the difference is between `parent_listener_` and `wrk_listener_`.

Reply:
Good idea. Done!